### PR TITLE
Revised calculator.py program structure

### PR DESCRIPTION
**Pull request by audreycode6**
# Description

Revised my calculator.py program. Included try/catch validation to user input (num1, num2) before moving to next step in program. Changed original input from user (num1, num2) to be through prompt(). Changed default case (case_) to use while loop & 'not' operator to validate operation input. 

# Changes
Change in structure of calculator.py program: 
- prompt() function to display messages to user VS original print()
- prompt('enter a number'), num1 = input() VS original num1 = int(input('Enter a number: ') to -- same for num2
- try/catch statement in invalid_input() function: if invalid input for num1/num2 return error and request input again VS original no input validation
- validate operation is correct option ('1', '2', '3', '4') with while loop and list of options VS original match-case default statement (case _: output = 'Error must enter 1, 2, 3, or 4)
- format printing results pretty with: if instance(output, str): print(output) VS original if type(output) != int: print(output)

# QA Instructions

_What do I need to do in order to test this change?_

1. ensure that the application works
2. feedback on PR & changes

### CODE REVIEW

File: lesson_2/.pylintrc
Comment (Doxify) [inline lesson_2/.pylintrc:1]: nit: things like this (adding pylint config) should be their own prs. Often times when you add a linter it will modify other files so its nice to have a single pr where you added the linter and updated each file.

File: lesson_2/calculator.py [hunk 1]
```diff
@@ -1,16 +1,44 @@
...
 
-# no error for input/ will crash if input is wrong --> would add error if was function
-num1 = int(input('Enter a number: '))
-num2 = int(input('Enter another number: '))
+def invalid_input(str_num):
+    try:
+        int(str_num)
+    except ValueError:
+        return True
+    return False
```
Comment (Doxify) [inline lesson_2/calculator.py:14]: praise: this is great, first time seeing a try/except from you and I like the function name! I can deduce what it does just by reading the function name. If you want to take it a step further, you could name it `is_invalid_input` so that it flows nicer in your code:

```py
if invalid_input(foo):
```

vs.

```py
if is_invalid_input(foo):
```

The below example is more like natural english which reduces cognitive load when someone is reading it for the first time.

File: lesson_2/calculator.py [hunk 2]
```diff
@@ -1,16 +1,44 @@
...
-operation = input('What operation would you like to perform?\n'
-                  '1) Addition 2) Subtraction 3) Multiplication 4) Division: ')
+prompt('Welcome to Calculator!')
+
+# NUM1
+prompt('Enter a number!')
+num1 = input()
+while invalid_input(num1):
+    prompt("Error: That isn't a valid number, expecting an integer!")
+    num1 = input()
```
Comment (Doxify) [inline lesson_2/calculator.py:23]: suggestion: make this a function so you can reuse it for num2

Anytime you find yourself copy/pasting code or re-writing the same code you should reach for a function. It will clean up this code and make it more readable.

File: lesson_2/calculator.py [hunk 3]
```diff
@@ -1,16 +1,44 @@
...
+num1 = int(num1) # ugly conversion to int, but following tut they did
+# -- def know i can make better
```
Comment (Doxify) [inline lesson_2/calculator.py:25]: nit: refrain from committing these comments to the code, instead add a comment to this PR. It's common practice for you to open a PR and then add comments explaining your decisions to reviewers. If you commit these comments it will just clutter this code. If someone wants to know why you made a decision they can just reference the PR in the future and see your comment.

File: lesson_2/calculator.py [hunk 4]
```diff
@@ -24,12 +52,9 @@
             output = 'Error: Cannot divide by zero!'
         else:
             output = num1 / num2
```
Comment (Doxify) [inline lesson_2/calculator.py:54]: nit: Same suggestion here about making this a function, it will make this code cleaner/more readable.

**Review by Doxify — APPROVED** [review]
LGTM

File: lesson_2/.pylintrc
Comment (audreycode6) [reply lesson_2/.pylintrc:1]: I'm confused by this comment; when you say their own prs, do you mean branch? how does it alter other files? I thought it was good practice to use a linter to check your code, so I added it to my env? @Doxify

File: lesson_2/calculator.py (hunk 2 above)
Comment (audreycode6) [reply lesson_2/calculator.py:23]: too many moving parts and going down a rabbit hole, so it's noted but I wont be adding the function @Doxify

File: lesson_2/calculator.py (hunk 4 above)
Comment (audreycode6) [reply lesson_2/calculator.py:54]: not clear what you are referencing should be in a function? the output = num1 (operator) num2 or just line ? @Doxify
